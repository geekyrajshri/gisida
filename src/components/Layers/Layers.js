require('./Layers.scss');
import Layer from '../Layer/Layer'

const Layers = ({ mapTargetId, layers, layerData, onLayerChange = f => f}) =>  
 <ul className="layers">
    {layers.map((layer, i) => 
     <Layer onLayerChange={onLayerChange} key={ i } mapTargetId={ mapTargetId } layer={ layer } layerData={ layerData } />)
  } 
 </ul>

export default Layers
