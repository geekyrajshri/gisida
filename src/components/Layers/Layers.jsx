import React from 'react';
import PropTypes from 'prop-types';
import Layer from '../Layer/Layer';

const Layers = ({ mapTargetId, layers, layerData, onLayerChange = f => f }) =>
  (<ul className="layers">
    {layers.map(layer =>
      (<Layer
        onLayerChange={onLayerChange}
        key={layer}
        mapTargetId={mapTargetId}
        layer={layer}
        layerData={layerData}
      />))
  }
  </ul>);

Layers.propTypes = {
  mapTargetId: PropTypes.string.isRequired,
  layers: PropTypes.arrayOf(PropTypes.any).isRequired,
  layerData: PropTypes.objectOf(PropTypes.any).isRequired,
  onLayerChange: PropTypes.func.isRequired,
};

export default Layers;
