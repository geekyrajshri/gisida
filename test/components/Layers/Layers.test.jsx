import React from 'react';
import renderer from 'react-test-renderer';
import Layers from '../../../src/components/Layers/Layers';
import layerData from '../../fixtures/layers.json';

describe('Layers', () => {
  const layers = ['test_layer'];
  const onLayerChange = jest.fn();

  it('Layers renders indicators', () => {
    const component = renderer.create(
      <Layers
        onLayerChange={onLayerChange}
        mapTargetId={'map-1'}
        layers={layers}
        layerData={layerData}
        sector='sector'
        defaultView = 'test'
      />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
