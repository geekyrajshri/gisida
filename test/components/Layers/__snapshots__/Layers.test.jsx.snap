// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layers Layers renders indicators 1`] = `
<ul
  className="layers"
  style={
    Object {
      "display": "none",
    }
  }
>
  <li
    className="layer map-1"
  >
    <label
      className="label"
      htmlFor="test_layer"
    >
      <input
        data-layer="test_layer"
        id="test_layer"
        name="sector"
        onClick={[Function]}
        type="checkbox"
        value="test_layer"
      />
      Test Layer
    </label>
  </li>
</ul>
`;
